corehq/apps/api/odata/tests/test_serializers.py

```diff
@@ -239,3 +241,97 @@ def test_next_link_absent(self):
             ODataCaseSerializer.get_next_url(meta, api_path),
             None
         )
+
+
+class TestODataFormSerializer(SimpleTestCase):
+
+    def test_selected_column_included(self):
+        self.assertEqual(
+            ODataFormSerializer.serialize_forms_using_config(
+                [{'user_id': 'the-user-id'}],
+                FormExportInstance(
+                    tables=[
+                        TableConfiguration(
+                            columns=[
+                                ExportColumn(
+                                    label='user-id',
+                                    item=ExportItem(
+                                        path=[
+                                            PathNode(name='user_id')
+                                        ]
+                                    ),
+                                    selected=True,
+                                )
+                            ]
+                        )
+                    ]
+                )
+            ),
+            [{'user-id': 'the-user-id'}]
+        )
+
+    def test_unselected_column_excluded(self):
+        self.assertEqual(
+            ODataFormSerializer.serialize_forms_using_config(
+                [{'user_id': 'the-user-id'}],
+                FormExportInstance(
+                    tables=[
+                        TableConfiguration(
+                            columns=[
+                                ExportColumn(
+                                    label='user-id',
+                                    item=ExportItem(
+                                        path=[
+                                            PathNode(name='user_id')
+                                        ]
+                                    ),
+                                    selected=False,
+                                )
+                            ]
+                        )
+                    ]
+                )
+            ),
+            [{}]
+        )
+
+    def test_missing_value_is_null(self):
+        self.assertEqual(
+            ODataFormSerializer.serialize_forms_using_config(
+                [{}],
+                FormExportInstance(
+                    tables=[
+                        TableConfiguration(
+                            columns=[
+                                ExportColumn(
+                                    label='user-id',
+                                    item=ExportItem(
+                                        path=[
+                                            PathNode(name='user_id')
+                                        ]
+                                    ),
+                                    selected=True,
+                                )
+                            ]
+                        )
+                    ]
+                )
+            ),
```

Comment:
These tests would be a little easier to read if you pulled the instance initialization out into a helper